### PR TITLE
Internal login JSON API

### LINKED ISSUE #2425: Admin login JSON API
We need an API for the admin client to exchange an FxA token for a JWT. This should function the same as the current login JSON API but it should use the admin FxA credentials and should not set any cookies.

Supports #2385.

### PR DESCRIPTION
**Pull request by mstriemer**
The login endpoint for the addons-frontend. Just a minor refactor of the old code and some copy/paste with slight modifications. There might be some clever way to avoid duplicating a few lines but this seems not-too-duplicated to me. Feel free to disagree.

Fixes #2425.

### CODE REVIEW

File: src/olympia/internal_tools/views.py [hunk 1]
```diff
@@ -29,17 +35,32 @@ class InternalAddonSearchView(AddonSearchView):
...
+            host=config['oauth_host'],
             query=urlencode(query)))
+
+
+class LoginView(APIView):
+
+    @with_user(format='json', config='internal')
+    def post(self, request, user, identity, next_path):
+        if user is None:
+            return Response({'error': ERROR_NO_USER}, status=422)
```
Comment (EnTeQuAk) [inline src/olympia/internal_tools/views.py]: Can we use `rest_framework.status` shortcuts to make the status codes more readable? I don't think we do that anywhere currently but I just noticed.

**Comment by EnTeQuAk** [comment]
Looks good, just one minor thing.

There's a test failing though, let's fix them first :)

File: src/olympia/internal_tools/views.py (hunk 1 above)
Comment (mstriemer) [inline src/olympia/internal_tools/views.py]: There's no status for `422` in rest framework, unfortunately.

**Comment by mstriemer** [comment]
> There's a test failing though, let's fix them first :)

Oops, fixed now.

**Comment by EnTeQuAk** [comment]
cool, r+

File: src/olympia/internal_tools/views.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/internal_tools/views.py]: I see, well, then let's not use it :laughing:

**Pull request merged**